utils/stm-to-jitx.stanza

```diff
@@ -0,0 +1,276 @@
+    val power-props = stm-to-jitx $ power-pin(pin-properties)
+    
+    ;Determine if the pads numeric (Double -> Int) or named (Ref -> String).
+    val numeric-pads? = any?({pad(_) is Double}, rows(pin-properties))
+    val named-pads? = any?({pad(_) is String}, rows(pin-properties))
+    
+    ;If the pads are neither or both, then we are using malformed data.
+    if numeric-pads? and named-pads? :
+      fatal("Generated a micro-controller with both Int and Ref pads.") 
+    if not numeric-pads? and not named-pads? :
+      fatal("Generated a micro-controller without pads.") 
+    
+    ;Generate the pin-properties.
+    ;Cases are split between Int and Ref pads, and across generic and power properties usages.
+    if numeric-pads? :
+      pin-properties :
+        [pin:Ref | pads:Int ... | side:Dir | generic-pin:GenericPin | power-pin:PowerPin]
+        for row in rows(pin-properties) do :
+          val pin-ref = ref-string-to-ref(pin(row))
+          val pad-name = to-int(pad(row) as Double)
+          match(generic-props?(row), power-props?(row)) :
+            (g:True, p:True)   : [(pin-ref) | pad-name | side(row) | generic-props | power-props]
+            (g:True, p:False)  : [(pin-ref) | pad-name | side(row) | generic-props |      -     ]
+            (g:False, p:True)  : [(pin-ref) | pad-name | side(row) |       -       | power-props]
+            (g:False, p:False) : [(pin-ref) | pad-name | side(row) |       -       |      -     ]
+    else :
+      pin-properties :
+        [pin:Ref | pads:Ref ... | side:Dir | generic-pin:GenericPin | power-pin:PowerPin]
+        for row in rows(pin-properties) do :
+          val pin-ref = ref-string-to-ref(pin(row))
+          val pad-name = raw-string-to-ref(pad(row) as String)
+          match(generic-props?(row), power-props?(row)) :
+            (g:True, p:True)   : [(pin-ref) | (pad-name) | side(row) | generic-props | power-props]
+            (g:True, p:False)  : [(pin-ref) | (pad-name) | side(row) | generic-props |      -     ]
+            (g:False, p:True)  : [(pin-ref) | (pad-name) | side(row) |       -       | power-props]
+            (g:False, p:False) : [(pin-ref) | (pad-name) | side(row) |       -       |      -     ]
+
+;Generate all bundles and store them in a table indexed by name.
+public defn to-jitx-bundles (bundles:Tuple<STMBundle>) -> HashTable<String, Bundle> :
+  val bundle-table = HashTable<String, Bundle>()
+  inside pcb-component :
+    for bundle-name in map(name, bundles) do :
+      pcb-bundle b :
+        name = bundle-name
+        pin p
+      bundle-table[bundle-name] = b
+  bundle-table
+
+;Generate all support statements inside of a component which will include bundles from bundle-table.
+public defn to-jitx-supports (supports:Tuple<STMSupports>, bundle-table:HashTable<String, Bundle>) -> False :
+  ;Generate a supports statement for a bundle with the given mappings.
+  defn generate-support (b:Bundle, mappings:Tuple<STMSupportMapping>) :
```

Comment:
@CuppoJava I really enjoyed dynamically generating JITX. Creating support statements around line 84 is especially neat, where I surprised myself in what we're able to achieve. Thanks for the pointers earlier!